### ISSUE TITLE
prefer-optional-chain: false positive on tricky condition

### ISSUE THREAD

**Issue by christophehurpeau** [oxc-project/tsgolint#829]
Here is the condition:

```
const isSameProjectAsPreviousIntervention = Boolean(
      intervention.projectId &&
      previousIntervention?.projectId?.toString() ===
        intervention.projectId.toString(),
    );
```

currently, typescript-eslint does not report any issue, which is correct. tsgolint report an issue:

```
  × typescript-eslint(prefer-optional-chain): Prefer using an optional chain expression instead, as it's more concise and easier to read.

 80 │         const isSameProjectAsPreviousIntervention = Boolean(
 81 │ ╭─▶       intervention.projectId &&
 82 │ │         previousIntervention?.projectId?.toString() ===
 83 │ ╰─▶         intervention.projectId.toString(),
 84 │         );
    ╰────
```

If written like this:

```
const isSameProjectAsPreviousIntervention =       previousIntervention?.projectId?.toString() ===
        intervention?.projectId?.toString();
```

and if both `intervention.projectId` and `previousIntervention?.projectId` are `null`, `isSameProjectAsPreviousIntervention` is true, but it should be false.

**Comment by camc314** [comment]
Thanks for the report - 0.18.0 is releasing now with a fix.
